pkg/workflow/tasks/custom/task.go

```diff
@@ -137,6 +137,7 @@ func (t *TaskLoader) makeTaskGenerator(templ string) (wfTypes.TaskGenerator, err
 			if err := json.Unmarshal(bt, &params); err != nil {
 				return nil, err
 			}
+			params["dependsOn"] = wfStep.DependsOn
```

Comment:
So here it is injecting dependsOn to the parameters of the WorkflowStep? What is it for?

Reply:
I'm also confused here. Is there any handle for `inputs/outputs` btw?

Reply:
the dependsOn is declared in compoment, and add it to param, so as to the provider handler `component-apply` can get the dependsOn from params

Reply:
If so, we should handle `inputs/outputs` here as well.